test.sh

```diff
@@ -40,6 +40,8 @@ for engine in "$@"; do
     done
 done
 
-__log_and_run cargo nextest run \
+nextest_run="nextest run \
     --all-targets --all-features --failure-output=never \
-    -- "${@/#/test_run::engine_}"
+    -- "${@/#/test_run::engine_}""
+
+__log_and_run ${nextest_run} || __log_and_run cargo ${nextest_run}
```

Comment:
There's an issue with quoting here. I'd also rewrite it this way to avoid running nextest twice when tests fail:

```bash
nextest_run=(
    nextest run
    --all-targets --all-features --failure-output=never
    -- "${@/#/test_run::engine_}"
    )

if command -v cargo-nextest &> /dev/null; then
    __log_and_run cargo "${nextest_run[@]}"
else
    __log_and_run "${nextest_run[@]}"
fi
```

Reply:
done, thanks